tutorials/javaFxPart3.md

```diff
@@ -15,38 +15,40 @@ JavaFX has an _event-driven architecture style_. As such, we programmatically de
 
 For Duke, there are two events that we want to respond to, namely the user pressing `Enter` in the `TextField` and left-clicking the `Button`. These are the `onAction` event for the `TextField` and the `onMouseClicked` event for the `Button`.
 
-For now, let’s have the application add a new `Label` with the text from the `TextField`. Update the `Main` class as follows. You'll need to add an `import javafx.scene.control.Label;` too.
+For now, let’s have the application add a new `DialogBox` with the text from the `TextField`, so the user can see the message they have sent. Update the `Duke` class as follows.
 ```java
 @Override
 public void start(Stage stage) {
-    // Step 1 code here
+    //Step 1 code here
 
     //Step 2 code here
+    //Delete or comment off the following section
+    //Label text = new Label("Hello!");
+    //ImageView displayPicture = new ImageView(user);
+    //DialogBox dialogBox = new DialogBox(text, displayPicture);
+    //dialogContainer.getChildren().addAll(dialogBox);
 
     //Step 3. Add functionality to handle user input.
     sendButton.setOnMouseClicked((event) -> {
-        dialogContainer.getChildren().add(getDialogLabel(userInput.getText()));
-        userInput.clear();
+        handleUserInput();
     });
 
     userInput.setOnAction((event) -> {
-        dialogContainer.getChildren().add(getDialogLabel(userInput.getText()));
-        userInput.clear();
+        handleUserInput();
     });
 }
 
 /**
  * Iteration 1:
- * Creates a label with the specified text and adds it to the dialog container.
- * @param text String containing text to add
- * @return a label with the specified text that has word wrap enabled.
+ * Creates a dialog box containing user input, and appends it to
+ * the dialog container. Clears the user input after processing.
  */
-private Label getDialogLabel(String text) {
-    // You will need to import `javafx.scene.control.Label`.
-    Label textToAdd = new Label(text);
-    textToAdd.setWrapText(true);
-
-    return textToAdd;
+private void handleUserInput() {
+        Label userText = new Label(userInput.getText());
+        dialogContainer.getChildren().addAll(
+            new DialogBox(userText, new ImageView(user))
+        );
+        userInput.clear();
```

Comment:
any reason why there's a change from `getDialogLabel` to `handleUserInput`? 
This current version handles both creation of the UI and also the processing of the input, which I feel could be better structured if we applied the separation of concerns design principle

Reply:
Sorry for the confusion. In the original tutorial, the getDialogLabel method was from iteration 1, and was later changed to handleUserInput in iteration 2. As the new version is no longer separated to iteration 1 and 2, I chose to follow iteration 2 for this part because it uses a DialogBox. Your concern is valid though. I chose to keep to handleUserInput because realistically the two methods only differ in one line (userInput.clear()). What do you think?